Update POCO docs to add OriginalNameAttribute (#182)
MicrosoftDocs/OData-docs: Odata-docs/client/using-poco-classes.md

---
title: "Using POCO classes"

> description: "This tutorial describes how to use POCO classes on the OData Client."

Reviewer: ```suggestion
description: "This tutorial describes how to use POCO classes in the OData Client."
```

Reviewer: > 

in/with can be used interchangeably here depending on the context

Author: #Resolved

author: mumbi-o
ms.author: mowambug
ms.date: 7/1/2019
ms.topic: article
 
---
# Using POCO classes

**Applies To**: [!INCLUDE[appliesto-odataclient](../includes/appliesto-odataclient-v7.md)]

## Using with POCO's

> While developing, you may want to use OData Client without a code generation tool like _ODataConnectedService_. In this case, one can still use the OData Client

Reviewer: ```suggestion
While developing, you may want to use the OData Client without a code generation tool like _ODataConnectedService_. In this case, one can still use the OData Client
```

Author: #Resolved

to access their service.

> In order to do this you simply extend the `DataServiceContext` class in your code and add some convenience methods as shown below:

Reviewer: ```suggestion
In order to do this you only need to extend the `DataServiceContext` class in your code and add some convenience methods as shown below:
```

Author: #Resolved

```csharp
// Person class with Odata annotations
[Key('Id')]
public class Person
{
        public string Id { get; set; }
        public string UserName { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
}

// Dataservice context
 public class Container : DataServiceContext
{
    public Container(Uri serviceRoot) : base(serviceRoot)
    {
              this.People = base.CreateQuery<Person>("People");
    }

    public DataServiceQuery<Person> People { get; }

    public void AddToPeople(Person person)
    {
       base.AddObject("People", person);
    }
}

```

> To add new objects to the peoples entity set use `context.AddObject("People", person)` or create a convenience method in the class to do the same.

Reviewer: ```suggestion
To add new objects to the "People" EntitySet use `context.AddObject("People", person)` or create a convenience method in the class to do the same.
```

Author: #Resolved

### Loading the service model

> This example allows you to use the client while using the current metadata which is fetched from network. In some cases this is not convenient and to override this behavior you need to change the constructor of the DataService context to include an action that can be used to fetch the model. Codegen tools do this by downloading the metadata and storing it in a verbatim string or a separate file which is then read upon instantiation of the context.

Reviewer: ```suggestion
This example allows you to use the client while using the current metadata which is fetched from the network. In some cases this is not convenient and to override this behavior you need to change the constructor of the DataService context to include an action that can be used to fetch the model. Codegen tools do this by downloading the metadata and storing it in a verbatim string or in a separate file which is then read upon instantiation of the context.
```

Author: #Resolved

To provide the same for our code we need to do the following.

```csharp
    // ctor from above
    public Container(Uri serviceRoot) : base(serviceRoot)
    {
        this.People = base.CreateQuery<Person>("People");
        // add the following lines
        this.Format.LoadServiceModel = () => util.GetEdmModel()  /* user action that returns a valid IEdmModel instance */  

>         this.Format.UseJson(); /* this instruction causes the model to be loaded instantly else the model is loaded lazily and cached when its needed */

Reviewer: ```suggestion
        this.Format.UseJson(); /* this instruction causes the model to be loaded instantly otherwise the model is loaded lazily and cached when its needed */
```

Author: #Resolved

    }

```

> Using the above method one can switch the model they want to use useful especially when writing tests.

Reviewer: `to use useful`?

Author: #Resolved

**Note**: Using the OData Connected Service is the recommended way to utilize all the new features in the OData Client.

### Using OriginalNameAttribute
The [OriginalNameAttribute](/dotnet/api/microsoft.odata.client.originalnameattribute) allows one to define an entity model with different property names from the OData service model.

> The [OriginalName](/dotnet/api/microsoft.odata.client.originalnameattribute.originalname) property denotes the original name of a property defined in OData service metadata.

Reviewer: ```suggestion
The [OriginalName](/dotnet/api/microsoft.odata.client.originalnameattribute.originalname) property denotes the original name of a property defined in the OData service metadata.
```

Author: #Resolved
